app/src/main/java/com/mapbox/navigation/examples/junctions/ShowJunctionsActivity.kt

```diff
@@ -0,0 +1,295 @@
+
+    private val locationObserver = object : LocationObserver {
+        override fun onNewRawLocation(rawLocation: Location) {}
+        override fun onNewLocationMatcherResult(locationMatcherResult: LocationMatcherResult) {
+            navigationLocationProvider.changePosition(
+                locationMatcherResult.enhancedLocation,
+                locationMatcherResult.keyPoints,
+            )
+            updateCamera(locationMatcherResult.enhancedLocation)
+        }
+    }
+
+    private val routesObserver = RoutesObserver { result ->
+        if (result.navigationRoutes.isNotEmpty()) {
+            CoroutineScope(Dispatchers.Main).launch {
+                routeLineApi.setNavigationRoutes(listOf(result.navigationRoutes[0])).apply {
+                    routeLineView.renderRouteDrawData(mapboxMap.getStyle()!!, this)
+                }
+            }
+            startSimulation(result.navigationRoutes[0])
+        }
+    }
+
+    private val routeProgressObserver = RouteProgressObserver { routeProgress ->
+        routeArrowApi.addUpcomingManeuverArrow(routeProgress).apply {
+            routeArrowView.renderManeuverUpdate(mapboxMap.getStyle()!!, this)
+        }
+    }
+
+    private val bannerInstructionsObserver = BannerInstructionsObserver { bannerInstructions ->
+        // The junction component is driven by banner instructions updates.
+        // Passing the instructions to the MapboxJunctionApi generates the data
+        // for updating the view.
+        junctionApi.generateJunction(bannerInstructions, junctionCallback)
+    }
+
+    private fun init() {
+        initNavigation()
+        initStyle()
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun initNavigation() {
+        mapboxNavigation = MapboxNavigationProvider.create(
+            NavigationOptions.Builder(this)
+                .accessToken(getString(R.string.mapbox_access_token))
+                .locationEngine(ReplayLocationEngine(mapboxReplayer))
+                .build()
+        )
+        mapboxNavigation.startTripSession()
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun initStyle() {
+        mapboxMap.loadStyleUri(Style.MAPBOX_STREETS) { style ->
+            routeLineView.initializeLayers(style)
+            binding.actionButton.setOnClickListener {
+                setDefaultRoute()
+                binding.actionButton.visibility = GONE
+            }
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun startSimulation(route: NavigationRoute) {
+        mapboxReplayer.stop()
+        mapboxReplayer.clearEvents()
+        mapboxReplayer.pushRealLocation(this, 0.0)
+        val replayEvents = ReplayRouteMapper().mapDirectionsRouteGeometry(route.directionsRoute)
+        mapboxReplayer.pushEvents(replayEvents)
+        mapboxReplayer.seekTo(replayEvents.first())
+        mapboxReplayer.play()
+    }
+
+    private fun findRoute(origin: Point, destination: Point) {
```

Comment:
If you look at other examples in this repository, you will see that we use hard-coded routes. The idea is to keep the examples simple and concise and only focus on the feature in hand. You can take a look at `ShowManeuversActivity` and how it uses `private val route = DirectionsRoute.fromJson(...)`.

Let's do the same here and remove this function